db/db_impl.cc

```diff
@@ -700,7 +721,9 @@ void DBImpl::BackgroundCall() {
 
   // Previous compaction may have produced too many files in a level,
   // so reschedule another compaction if needed.
-  MaybeScheduleCompaction();
+  if (!suspending_compaction_.load(std::memory_order_acquire)) {
```

Comment:
There's already a check for "suspending_compaction_.Acquire_Load()" in "MaybeScheduleCompaction", but it also checks for "imm_" being null. There are also other places where MaybeScheduleCompaction() might be called based on "suspending_compaction_". Wouldn't it be better to have that check in "MaybeScheduleCompaction()", whatever the value of "imm_ "is?

Reply:
@thierrybouchardskyboxlabs @lorenhoffman The places where `MaybeScheduleCompaction()` is wrapped which a `!suspending_compaction_` check is to make sure that compactions aren't suspending while inside `MaybeScheduleCompaction()` it's the opposite checking to make sure compactions are disabled. I think this works because not all the `MaybeScheduleCompaction()` calls are wrapped with `!suspending_compaction_` check. Something to note is this part of the pr is from the second commit which was added a while back to resolve a deadlock that was occurring, so I don't feel too comfortable fiddling around moving stuff around without fully understanding what was causing the deadlock. I'm just going to keep it as is for now, if this a major thing you think that should get resolved before it's merged, I'll take a deeper look into it!